File: Leetcode/October/Day1_Number_of_Recent_Calls.cpp
#include<queue>

class RecentCounter {
public:
    queue <int> q;
    RecentCounter() {
        
    }
    
    int ping(int t) {
        q.push(t);
        while(q.front() < t - 3000) // If q.front() is beyond lower limit -> pop it out
        {
            q.pop();
        }
        return q.size();
        
    }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */

File: LeetCode/Monthly Challenges/October_LeetCoding_Challenge/Day1_Number_of_Recent_Calls/Day1_Number_of_Recent_Calls.java
class RecentCounter {
    Queue<Integer> queue;

    public RecentCounter() {
        queue = new LinkedList<>();

    }

    public int ping(int t) {
        queue.add(t);
        while (queue.peek() < t - 3000)
            queue.poll();
        return queue.size();
    }
}
